crates/dbsp/src/circuit/runtime.rs

```diff
@@ -638,28 +560,51 @@ impl Runtime {
             .write()
             .map(|mut guard| *guard = Some(panic_info));
     }
+
+    /// Spawn a new thread using `builder` and `f`. If the current thread is
+    /// associated with a runtime, then the new thread will also be associated
+    /// with the same runtime and worker index.
+    pub(crate) fn spawn_background_thread<F>(builder: Builder, f: F) -> Thread
```

Comment:
I believe I also did it first like this but there was a problem with making sure the threads get removed again when the pipeline crashes/shuts down. Not sure we have a test for that. Maybe we should. Also can easily be that this isn't a problem anymore the way you set it up now.